src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.ts

```diff
@@ -338,24 +335,10 @@ export class FileUploadAssessmentComponent implements OnInit, OnDestroy {
         this.assessmentsAreValid = true;
         this.invalidError = undefined;
 
-        if ((!this.generalFeedback.detailText || this.generalFeedback.detailText.length === 0) && this.unreferencedFeedback && this.unreferencedFeedback.length === 0) {
-            this.totalScore = 0;
-            this.assessmentsAreValid = false;
-            return;
-        }
-
-        let credits = this.unreferencedFeedback.map((assessment) => assessment.credits);
-
-        if (!this.invalidError && !credits.every((credit) => credit && !isNaN(credit))) {
-            this.invalidError = 'artemisApp.fileUploadAssessment.error.invalidScoreMustBeNumber';
-            this.assessmentsAreValid = false;
-            credits = credits.filter((credit) => credit && !isNaN(credit));
-        }
+        const hasUnreferencedFeedback = Feedback.haveCreditsAndComments(this.unreferencedFeedback);
+        // When unreferenced feedback is set, it has to be valid (score + detailed text)
+        this.assessmentsAreValid = hasUnreferencedFeedback;
```

Comment:
Feedback.haveCreditsAndComments is more explanatory than hasUnreferencedFeedback. So I think it can be omitted. 

Like this.assessmentsAreValid = Feedback.haveCreditsAndComments(this.unreferencedFeedback);